src/components/ListHeader/ListHeader.js

```diff
@@ -1,26 +1,94 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faShareAlt } from '@fortawesome/free-solid-svg-icons';
+import { faShareAlt, faCopy, faPencilAlt, faClock, faTrash } from '@fortawesome/free-solid-svg-icons';
 import { ReactComponent as Logo } from '../../static/logo.svg';
 
 import styles from './styles.module.css';
 
-const ListHeader = () => (
-  <div className={styles.listHeader}>
-    <div className={styles.headerRow}>
-      <Logo className={styles.logo} title="Logo" />
-      <FontAwesomeIcon
-        icon={faShareAlt}
-        className={styles.shareIcon}
-        title="Share this list"
-        tabIndex={0}
-      />
-    </div>
-    <div className={styles.headerRow}>
-      <h3>Shopping List</h3>
-      <h4>15 items</h4>
+const ListHeader = () => {
+  const [listDetailsShown, showListDetails] = useState();
+
+  const shareList = () => {
+    if (navigator.share) {
+      navigator.share({
+        title: 'Share Your List!',
+        url: 'https:thelist.app',
+      });
+    } else {
+      const listText = document.getElementById('listText');
+      listText.select();
+      document.execCommand('copy');
+    }
+  };
+
+  return (
+    <div className={styles.listHeader}>
+      <div className={styles.headerRow}>
+        <Logo className={styles.logo} title="Logo" />
+        <FontAwesomeIcon
+          icon={faShareAlt}
+          className={styles.shareIcon}
+          title="Share this list"
+          tabIndex={0}
+          onClick={() => showListDetails(!listDetailsShown)}
+        />
+      </div>
+      <div className={styles.headerRow}>
+        <h3>
+          Shopping List
+          <FontAwesomeIcon
+            icon={faPencilAlt}
+            className={styles.pencilIcon}
+            title="Edit list name"
+            tabIndex={0}
+          />
+        </h3>
+        <h4>15 items</h4>
+      </div>
+      {listDetailsShown ? (
+        <>
+          <div className={styles.subheadingRow}>
+            <h4>
+              <FontAwesomeIcon
+                icon={faClock}
+                className={styles.clockIcon}
+              />
+              Created 1st Jan 2020
+            </h4>
+          </div>
+          <div className={styles.copyList}>
+            <h5>Share -</h5>
+            <input
+              type="text"
+              spellCheck="false"
+              value="https:thelist.app"
+              id="listText"
```

Comment:
I see what you mean about this not being very 'react', you can do the following:

1. Where you have the useState hook, add [another hook](https://reactjs.org/docs/hooks-reference.html#useref) which is a variable that will be a reference to this component: 
   ```js
   const listText = useRef(null);
   ```

1. For this input component, pass in the ref:
   ```jsx
   <input
     ref={listText}
     ...
   />
   ```

1. `listText.current` is now a reference to this component for you to do what you need, e.g. when you want to copy the text:
   ```js
   listText.current.select();
   document.execCommand('copy');
   ```

Reply:
One main advantage is that you're not relying on IDs being unique if this component gets re-used

Reply:
Ok cool thanks! I'll have a go at implementing this!